src/View/Helper/TextHelper.php

```diff
@@ -162,12 +162,43 @@ protected function _linkUrls(string $text, array $htmlOptions): string
             if (!preg_match('#^[a-z]+\://#i', $url)) {
                 $url = 'http://' . $url;
             }
+
+            $linkOptions = $htmlOptions;
+            unset($htmlOptions['maxLength'], $htmlOptions['stripProtocol'], $htmlOptions['ellipsis']);
+            $link = $this->_prepareLinkLabel($link, $linkOptions);
+
             $replace[$hash] = $envelope[0] . $this->Html->link($link, $url, $htmlOptions) . $envelope[1];
         }
 
         return strtr($text, $replace);
     }
 
+    /**
+     * Prepares the link label.
+     *
+     * ### Options
+     *
+     * - `stripProtocol` Strips http:// and https:// from the beginning of the link. Default off.
+     * - `maxLength` The maximum length of the link label. Default off.
+     * - `ellipsis` The string to append to the end of the link label. Defaults to UTF8 version.
+     *
+     * @param string $name Link label.
+     * @param array $options<string, mixed> $htmlOptions The options for the generated link label.
+     * @return string Modified link label.
+     */
+    protected function _prepareLinkLabel(string $name, array $options): string
+    {
+        if (isset($options['stripProtocol']) && $options['stripProtocol'] === true) {
+            $name = (string)preg_replace('(^https?://)', '', $name);
+        }
+        if (!empty($options['maxLength']) && mb_strlen($name) > $options['maxLength']) {
+            $name = mb_substr($name, 0, $options['maxLength']);
+            $name .= $options['ellipsis'] ?? '…';
```

Comment:
Shouldn't we trim off enough room for the ellipsis so the total length is maxLength?

Reply:
We could mb_strlen() it and remove more to have it fitting.
I just thought it already saves a lot more than before, so it seems an incremental win already.

Whatever the plenum is, I am fine either way.

Reply:
Good call, I looked into it again and think it would be consistent with other text helper truncation if we included it.